robotoff/__main__.py

```diff
@@ -31,16 +32,45 @@ def predict_insight(ocr_url: str):
 
         print(json.dumps(results, indent=4))
 
-    @click.command()
-    @click.argument("input_")
+    @click.command(
+        help="""
+            Generate OCR insights of the requested type.
+            
+            \b
+            SOURCE can be either:
+            * the path to a JSON file, a (gzipped-)JSONL file, or a directory
+              containing JSON files
+            * a barcode
+            * the '-' character: input is read from stdin and assumed to be JSONL
+              
+            Output is JSONL, each line containing the insights for one document.
+        """
+    )
+    @click.argument("source")
     @click.option("--insight-type", "-t", required=True)
-    @click.option("--output", "-o")
-    def generate_ocr_insights(input_: str, insight_type: str, output: str):
+    @click.option(
+        "--output",
+        "-o",
+        help="file to write output to, stdout if not specified",
+        type=click.Path(dir_okay=False, writable=True),
+    )
+    @click.option(
+        "--keep-empty/--no-keep-empty",
+        default=False,
+        help="keep documents with empty insight",
+        show_default=True,
+    )
+    def generate_ocr_insights(
+        source: str, insight_type: str, output: str, keep_empty: bool
+    ):
         from robotoff.cli import insights
         from robotoff.utils import get_logger
 
+        if source == "-":
+            source = sys.stdin
```

Comment:
mypy error here: Incompatible types in assignment (expression has type "TextIO", variable has type "str")